pyLOM/NN/utils.py

```diff
@@ -318,19 +467,298 @@ def get_splits(
+            for i in range(remainder):
+                idx_to_add_at = i % len(subset_lengths)
+                subset_lengths[idx_to_add_at] += 1
+            sizes = subset_lengths
+            for i, length in enumerate(sizes):
+                if length == 0:
+                    raiseWarning(
+                        f"Length of split at index {i} is 0. "
+                        f"This might result in an empty dataset."
+                    )
+
+        # Cannot verify that dataset is Sized
+        if sum(sizes) != len(self.parameters):
+            raiseError(
+                "Sum of input lengths does not equal the length of the number of parameters!"
+            )
+        if shuffle:
+            indices = randperm(sum(sizes), generator=generator).tolist()  # type: ignore[arg-type, call-overload]
+        else:
+            indices = list(range(sum(sizes)))
+        sizes = cast(Sequence[int], sizes)
+        split_datasets = []
+        for offset, length in zip(accumulate(sizes), sizes):
+            new_indices = []
+            for index in indices[offset - length : offset]:
+                vars_in_extended_indices = list(range(index * len(self.variables_in), (index + 1) * len(self.variables_in)))
+                new_indices.extend(vars_in_extended_indices)
+            if not return_views:
+                # Get the split data
+                if self.variables_in is not None:
+                    split_variables_in = self.variables_in
+                    # print(self.parameters.shape, self.parameters[offset - length : offset])
+                    split_parameters = self.parameters[indices[offset - length : offset]]
+                else:
+                    split_variables_in = None
+                    split_parameters = None
+                    
+                # Get the split output variables
+                split_variables_out = self.variables_out[new_indices]
+
+                # Create a new dataset instance with the split data
+                split_dataset = Dataset(
+                    variables_out=tuple([split_variables_out[:, i].reshape(-1, *self.mesh_shape).numpy()
+                                    for i in range(self.num_channels)]),
+                    mesh_shape=self.mesh_shape,
+                    variables_in=split_variables_in.numpy() if split_variables_in is not None else None,
+                    # Parameters must be transposed to pass only as many list as the number of variables
+                    parameters=split_parameters.T.tolist() if split_parameters is not None else None,
+                    combine_parameters_with_cartesian_prod=False,
+                    snapshots_by_column=False  # Since the data is already reshaped
+                )
+                split_datasets.append(split_dataset)
+            else:
+                split_datasets.append(Subset(self, new_indices))
+
+        return split_datasets
+    
+    @classmethod
+    def load_from_file(
```

Comment:
I forgot to change the tests sorry, i do it now. 
About the examples with the VAE, i made sure that they work, and the creation of the `pylom.NN.Dataset` seems quite simple on the examples, if you want, I can change it too.
and I rename the method too.

Reply:
i just realized that the examples with the VAE do a preprocessing
```
d    = pyLOM.Dataset.load(DSETDIR,ptable=m.partition_table)
u_x  = d['VELOX']
u_m  = pyLOM.math.temporal_mean(u_x)
u_xm = pyLOM.math.subtract_mean(u_x, u_m)
```
and simply calling `pyLOM.NN.Dataset.load` won't work, it would be needed to call `map` to process the data.

Reply:
I've run the KAN and MLP test and they pass.
However, the test of VAE_Cylinder_2D doesn't pass. I tried to run it on develop branch but it didn't pass either. The training loss decreases normally in both branches, maybe I'm missing something

Reply:
> Thanks @DavidRamosArchilla. I will check the issue with the VAE_Cylinder_2D and fix it.

if it is useful, i ran the test with this command
`python run_testsuite.py -m NN -f testsuite.json -t "VAE_Cylinder_2D,VAE_Channel_3D"`
in case I've done something wrong

Reply:
No no, you're good. The test should pass in the github system and I've check that it actually does. So there is nothing to worry about.